windows/vtpm_cold_add_remove/vtpm_add_remove_check.yml

```diff
@@ -32,16 +32,23 @@
   include_tasks: ../utils/win_get_tpm_status.yml
 
 # Check vTPM device exists in guest OS
-- name: "Make sure TPM device is present and ready in guest OS"
-  ansible.builtin.assert:
-    that:
-      - win_tpm_info['TpmPresent'] | bool
-      - win_tpm_info['TpmReady'] | bool
-      - win_tpm_info['TpmEnabled'] | bool
-      - win_tpm_info['TpmActivated'] | bool
-      - win_tpm_info['ManufacturerIdTxt'] | trim == 'VMW'
-    fail_msg: "TPM device status got in Windows guest OS: {{ win_tpm_info }}, which contains unexpected values."
+- name: "Check TPM device status in guest OS"
   when: vtpm_test_operation == 'add'
+  block:
+    - name: "Make sure TPM device is present and ready"
+      ansible.builtin.assert:
+        that:
+          - win_tpm_info['TpmPresent'] | bool
+          - win_tpm_info['TpmReady'] | bool
+          - win_tpm_info['ManufacturerIdTxt'] | trim == 'VMW'
+        fail_msg: "TPM device status got in Windows guest OS: {{ win_tpm_info }}, which contains unexpected values."
+    - name: "Make sure TPM device is enabled and activated"
+      ansible.builtin.assert:
+        that:
+          - win_tpm_info['TpmEnabled'] | bool
+          - win_tpm_info['TpmActivated'] | bool
+        fail_msg: "TPM device status got in Windows guest OS: {{ win_tpm_info }}, which contains unexpected values."
+      when: guest_os_ansible_distribution is not search('2016|2019')
```

Comment:
Can here check server version is 2022 or later?

Reply:
There is also Windows 10 and Windows 11.